resources/migrations/058_update_migrations.yaml

```diff
@@ -501,6 +501,373 @@ databaseChangeLog:
+                  name: api_key_id
+                  type: int
+                  remarks: FK to api key used by agent
+              - column:
+                  name: database_id
+                  type: int
+                  remarks: FK to db used for this workspace
+              - column:
+                  name: schema
+                  type: varchar(254)
+                  remarks: Isolated schema where work happens
+                  constraints:
+                    nullable: true
+              - column:
+                  name: graph
+                  type: ${text.type}
+                  remarks: Graph configuration serialized as JSON
+              - column:
+                  name: created_at
+                  type: ${timestamp_type}
+                  remarks: Timestamp when the workspace was created
+                  defaultValueComputed: current_timestamp
+              - column:
+                  name: updated_at
+                  type: ${timestamp_type}
+                  remarks: Timestamp when the workspace was last updated
+                  defaultValueComputed: current_timestamp
+      rollback:
+        - dropTable:
+            tableName: workspace
+
+  - changeSet:
+      id: v58.2025-11-18T12:00:01
+      author: piranha
+      comment: Add index on workspace.collection_id for quick lookups by collection
+      changes:
+        - createIndex:
+            tableName: workspace
+            indexName: idx_workspace_collection_id
+            columns:
+              - column:
+                  name: collection_id
+      rollback: # will be removed along the table
+
+  - changeSet:
+      id: v58.2025-11-18T12:00:02
+      author: piranha
+      comment: Add index on workspace.user_id for quick lookups by owner
+      changes:
+        - createIndex:
+            tableName: workspace
+            indexName: idx_workspace_user_id
+            columns:
+              - column:
+                  name: user_id
+      rollback: # will be removed along the table
+
+  - changeSet:
+      id: v58.2025-11-18T12:00:03
+      author: piranha
+      comment: Add index on workspace.api_key_id for quick lookups by agent key
+      changes:
+        - createIndex:
+            tableName: workspace
+            indexName: idx_workspace_api_key_id
+            columns:
+              - column:
+                  name: api_key_id
+      rollback: # will be removed along the table
+
+  - changeSet:
+      id: v58.2025-11-18T12:00:04
+      author: piranha
+      comment: Add index on workspace.database_id for quick lookups by database
+      changes:
+        - createIndex:
+            tableName: workspace
+            indexName: idx_workspace_database_id
+            columns:
+              - column:
+                  name: database_id
+      rollback: # will be removed along the table
+
+  - changeSet:
+      id: v58.2025-11-18T12:00:05
+      author: piranha
+      comment: Add workspace_id column to metabase_database table
```

Comment:
This column doesn't make sense to me - we'll be able to have multiple workspaces for a given DWH